geonode/geoserver/helpers.py

```diff
@@ -407,17 +407,11 @@ def set_dataset_style(saved_dataset, title, sld, base_file=None):
         except Exception as e:
             logger.exception(e)
     else:
-        style = gs_catalog.get_style(saved_dataset.name, workspace=saved_dataset.workspace) or \
-            gs_catalog.get_style(saved_dataset.name)
         try:
-            if not style:
-                style = gs_catalog.create_style(
-                    saved_dataset.name, sld,
-                    overwrite=True, raw=True,
-                    workspace=saved_dataset.workspace)
-            elif sld:
-                style.style_format = _extract_style_version_from_sld(sld)
-                style.update_body(sld)
```

Comment:
Not sure we want to remove this check. @giohappy ?

Reply:
based on what I saw, that update_body call geoserver again with the same call of the create_style. Imo is a duplicated call

Reply:
@mattiagiupponi as far as I remember that was invoked in order to correctly set the `style_format` on GeoServer which was causing some issues later on, when trying to update it.

Reply:
@afabiani @mattiagiupponi why don't we set the style_format parameter for `create_style`, in case `sld` is defined, instead of issuing the `update_body` afterward?

Reply:
imo the call is still redundant. With some tests, it looks like the update_body does not update the resource body in GS. By calling just with the create_style in override, everything works fine. 
In the end, the two calls are the same, so all that code is a bit redundant

Reply:
`style_format` check readded as parameter in the call

Reply:
@mattiagiupponi what if the style already exists in the GeoServer catalog?

Reply:
it has `overwrite=True` which means that the sld provided will override the existing one.

Reply:
well, it can throw an exception if the sld provided is invalid from geoserver point of view. I that case we log it